src/lib/services/segment-service.ts

```diff
@@ -115,6 +122,17 @@ export class SegmentService {
         await this.segmentStore.removeFromStrategy(id, strategyId);
     }
 
+    async validateName(name: string): Promise<void> {
+        if (!name) throw new BadDataError('Segment name is not valid');
+
+        try {
+            await this.segmentStore.getByName(name);
+        } catch (error) {
+            return;
```

Comment:
Is it possible to find another way to check this? My worry here is that if there is a legitimate error that bubbles up from getByName (e.g. someone drops the table) this will hide that error case. Some of the other stores use an 'exists' method  for this kind of thing, might be worth looking into

Reply:
Good idea, we create an existsByName method for this purpose and it looks a bit nicer and cleaner.